src/java.base/share/classes/java/lang/invoke/MethodHandleProxies.java

```diff
@@ -162,6 +204,11 @@ public static <T> T asInterfaceInstance(final Class<T> intfc, final MethodHandle
             throw newIllegalArgumentException("not a public interface", intfc.getName());
         if (intfc.isSealed())
             throw newIllegalArgumentException("a sealed interface", intfc.getName());
+        if (intfc.isHidden())
+            throw newIllegalArgumentException("a hidden interface", intfc.getName());
+        if (!VM.isModuleSystemInited())
```

Comment:
I don't expect this is needed.  I assume you are thinking for LMF to use this API?

Reply:
Proxy-based impl had this check, so I'd assume MHP might want to defend against the same kind of improper usage.

Reply:
Proxy is used by annotation implementation.   I suspect that's why that check was there.

For `MHP::asInterfaceInstance`, I would drop the `isModuleSystemInitied` check until the API is being used during early startup.   I don't expect it be called before module system is initialized.